File: packages/sdk/src/models/userinfo.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

/**
 * UserInfo with user displayName, objectId and preferredUserName.
 */
export interface UserInfo {
  /**
   * User Display Name.
   *
   * @readonly
   */
  displayName: string;

  /**
   * User unique reference within the Azure Active Directory domain.
   *
   * @readonly
   */
  objectId: string;

  /**
   * User tenant ID.
   *
   * @readonly
   */
  tenantId: string;

  /**
   * Usually be the email address.
   *
   * @readonly
   */
  preferredUserName: string;
}

export interface UserTenantIdAndLoginHint {
  /**
   * User Login Hint.
   *
   * @readonly
   */
  loginHint: string;

  /**
   * User Tenant Id.
   *
   * @readonly
   */
  tid: string;
}

File: packages/dotnet-sdk/src/TeamsFx/Model/UserInfo.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
namespace Microsoft.TeamsFx;

/// <summary>
/// UserInfo with user displayName, objectId and preferredUserName.
/// </summary>
public class UserInfo
{
    /// <summary>
    /// User Display Name.
    /// </summary>
    public string DisplayName { get; set; }

    /// <summary>
    /// User unique reference within the Azure Active Directory domain.
    /// </summary>
    public string PreferredUserName { get; set; }

    /// <summary>
    /// Usually be the email address.
    /// </summary>
    public string ObjectId { get; set; }
}
